spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -149,6 +149,58 @@ describe("Router", function()
     end)
   end)
 
+  describe("URI arguments (querystring)", function()
+
+    setup(function()
+      insert_apis {
+        {
+          name = "api-1",
+          upstream_url = "http://httpbin.org",
+          hosts = { "example.com" },
+        },
+      }
+
+      assert(helpers.start_kong())
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("preserves URI arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get",
+        query  = {
+          foo   = "bar",
+          hello = "world",
+        },
+        headers = {
+          ["Host"] = "example.com",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.equal("bar", json.args.foo)
+      assert.equal("world", json.args.hello)
+    end)
+
+    it("does not proxy '$is_args' if URI does not contain arguments", function()
```

Comment:
Something like this patch resolves the issue from what I can tell: https://gist.github.com/p0pr0ck5/e742a478a8bfcaeabeefddafd54af947

However, the test in question fails, assumedly because of upstream normalization (both httpbin and mockbin strip `?` when no query args are present). tshark proof: https://gist.github.com/p0pr0ck5/d19fb99c88ed5e2b1efdc53707912bb1#file-gistfile1-txt-L375